python/tests/integration_tests/test_runs.py

```diff
@@ -455,3 +455,28 @@ async def my_async_generator(num: int) -> AsyncGenerator[str, None]:
             ]
         )
     }
+
+
+async def test_end_metadata_with_run_tree(langchain_client: Client):
+    project_name = "__My Tracer Project - test_end_metadata_with_run_tree"
+    run_meta = uuid.uuid4().hex
+
+    run_tree = RunTree(
+        name="my_chain_run",
+        run_type="chain",
+        project_name=project_name,
+    )
+
+    run_tree.end(metadata={"final_metadata": run_meta}, outputs={"result": "success"})
+    run_tree.post()
+
+    filter_ = (
+        f'and(eq(metadata_key, "final_metadata"), eq(metadata_value, "{run_meta}"))'
+    )
+    poll_runs_until_count(langchain_client, project_name, 1, filter_=filter_)
+
+    runs_ = list(langchain_client.list_runs(project_name=project_name, filter=filter_))
```

Comment:
I would explicitly set the run_id in `RunTree`, then fetch the run by run id. That guarantees you're looking at the exact same run you just posted. If someone else were to write to this project, it might throw off your test